### PR TITLE
Allowing RNW to refer to alternate JSI truths

### PR DESCRIPTION
**Pull request by mganandraj**
This change will enable the following,
1. Centralized control of V8 and Hermes package versions
2. Allowing RNW to consume alternate JSI headers and sources. Currently, we include the JSI source along with the V8 and Hermes packages .. This change will ensure that we use the right version of JSI source when consuming these packages.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3590)

### CODE REVIEW

File: vnext/PropertySheets/React.Cpp.props [hunk 1]
```diff
@@ -17,15 +17,26 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <USE_HERMES Condition="'$(USE_HERMES)' == ''">false</USE_HERMES>
+    <USE_HERMES Condition="'$(USE_HERMES)' == ''">true</USE_HERMES>
```
Comment (mganandraj) [inline vnext/PropertySheets/React.Cpp.props]: I'll revert USE_HERMES to false before submitting.

File: vnext/ReactWindowsCore/packages.config [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="boost" version="1.68.0.0" targetFramework="native" />
-  <package id="ReactNative.Hermes.Windows" version="0.1.3" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
-  <package id="ReactNative.V8JSI.Windows" version="0.1.4" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_V8)' == 'true'"/>
+  <package id="ReactNative.Hermes.Windows" version="$(HERMES_Version)" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
+  <package id="ReactNative.V8JSI.Windows" version="$(V8_Version)" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_V8)' == 'true'"/>
```
Comment (tudorms) [inline vnext/ReactWindowsCore/packages.config]: >V8_Version [](start = 53, length = 10)

Does this break the CLI nuget command? I'm wondering if CI will be happy with it.

File: vnext/ReactUWP/packages.config [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="boost" version="1.68.0.0" targetFramework="native" />
-  <package id="ReactNative.Hermes.Windows" version="0.1.3" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
-  <package id="ReactNative.V8JSI.Windows" version="0.1.4" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_V8)' == 'true'" />
+  <package id="ReactNative.Hermes.Windows" version="$(HERMES_Version)" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_HERMES)' == 'true'" />
+  <package id="ReactNative.V8JSI.Windows" version=version="$(V8_Version)" targetFramework="native" Condition="'$(OSS_RN)' != 'true' AND '$(USE_V8)' == 'true'" />
```
Comment (stecrain) [inline vnext/ReactUWP/packages.config]: typo here

File: vnext/Desktop.ABITests/React.Windows.Desktop.ABITests.vcxproj [hunk 4]
```diff
@@ -52,7 +52,7 @@
   <ItemDefinitionGroup>
     <ClCompile>
       <SDLCheck>true</SDLCheck>
-      <AdditionalIncludeDirectories>$(VCInstallDir)UnitTest\include;$(ReactNativeDir)\ReactCommon;$(ReactNativeDir)\ReactCommon\jsi;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(VCInstallDir)UnitTest\include;$(ReactNativeDir)\ReactCommon;$(JSI_Source);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (stecrain) [inline vnext/Desktop.ABITests/React.Windows.Desktop.ABITests.vcxproj]: Now that RuntimeHolder.h is included here I think you need to remove RuntimeHolder.h from JSI.Shared.vcxitems and update those includes

File: vnext/Desktop.IntegrationTests/React.Windows.Desktop.IntegrationTests.vcxproj [hunk 5]
```diff
@@ -40,7 +40,7 @@
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup>
-    <IncludePath>$(ReactNativeWindowsDir);$(ReactNativeWindowsDir)Common;$(ReactNativeWindowsDir)stubs;$(FollyDir);$(ReactNativeDir)\ReactCommon;$(ReactNativeDir)\ReactCommon\jsi;$(ReactNativeWindowsDir)ReactWindowsCore;$(ReactNativeWindowsDir)include\ReactWindowsCore;$(ReactNativeWindowsDir)Desktop;$(ReactNativeWindowsDir)IntegrationTests;$(MSBuildProjectDirectory);$(IncludePath)</IncludePath>
+    <IncludePath>$(ReactNativeWindowsDir);$(ReactNativeWindowsDir)Common;$(ReactNativeWindowsDir)stubs;$(FollyDir);$(ReactNativeDir)\ReactCommon;$(JSI_Source);$(ReactNativeWindowsDir)ReactWindowsCore;$(ReactNativeWindowsDir)include\ReactWindowsCore;$(ReactNativeWindowsDir)Desktop;$(ReactNativeWindowsDir)IntegrationTests;$(MSBuildProjectDirectory);$(IncludePath)</IncludePath>
```
Comment (JunielKatarn) [inline vnext/Desktop.IntegrationTests/React.Windows.Desktop.IntegrationTests.vcxproj]: (nit)

MSBuild properties usually follow Pascal case convention.
For better readability, I suggest to rename this macro to either `JSISource` or `JsiSource`.
More over, to match current conventions, I'd personally rename it as `JSIDir` or `JSISourceDir`.

File: vnext/PropertySheets/React.Cpp.props [hunk 6]
```diff
@@ -18,14 +18,25 @@
 
   <PropertyGroup>
     <USE_HERMES Condition="'$(USE_HERMES)' == ''">false</USE_HERMES>
+    <HERMES_Version Condition="'$(HERMES_Version)' == ''">0.1.6</HERMES_Version>
```
Comment (JunielKatarn) [inline vnext/PropertySheets/React.Cpp.props]: Is Hermes to be used throughout all RNW?
If not, I suggest creating a Hermes-specific `props` file and either:
1. Include it right after `React.Cpp.props` in the appropriate VCXPROJ files.
2. Wrap `React.Cpp.props` and include it in the appropriate VCXPROJ files.

Comment (mganandraj) [reply vnext/PropertySheets/React.Cpp.props]: Julio.. thanks for the suggestion.. I'll do it in a subsequent change..

File: vnext/Desktop.ABITests/React.Windows.Desktop.ABITests.vcxproj (hunk 4 above)
Comment (mganandraj) [reply vnext/Desktop.ABITests/React.Windows.Desktop.ABITests.vcxproj]: Done.

**Pull request merged**